tests/test_persistence.py

```diff
@@ -103,41 +124,450 @@ def test_update_with_binance(limit_buy_order, limit_sell_order, fee, caplog):
+            binance -1x: (1-(66.1663784375 / 59.85)) * 1 = -0.1055368159983292
+            binance -3x: (1-(66.1663784375 / 59.85)) * 3 = -0.3166104479949876
+            kraken  -1x: (1-(66.2311650 / 59.85)) * 1    = -0.106619298245614
+            kraken  -3x: (1-(66.2311650 / 59.85)) * 3    = -0.319857894736842
+    """
+
+    trade = Trade(
+        id=2,
+        pair='ETH/BTC',
+        stake_amount=60.0,
+        open_rate=2.0,
+        amount=30.0,
+        is_open=True,
+        open_date=arrow.utcnow().datetime,
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
     )
     assert trade.open_order_id is None
     assert trade.close_profit is None
     assert trade.close_date is None
 
     trade.open_order_id = 'something'
-    trade.update(limit_buy_order)
+    trade.update(limit_buy_order_usdt)
     assert trade.open_order_id is None
-    assert trade.open_rate == 0.00001099
+    assert trade.open_rate == 2.00
     assert trade.close_profit is None
     assert trade.close_date is None
     assert log_has_re(r"LIMIT_BUY has been fulfilled for Trade\(id=2, "
-                      r"pair=ETH/BTC, amount=90.99181073, open_rate=0.00001099, open_since=.*\).",
+                      r'pair=ETH/BTC, amount=30.00000000, '
+                      r"is_short=False, leverage=1.0, open_rate=2.00000000, open_since=.*\).",
                       caplog)
 
     caplog.clear()
     trade.open_order_id = 'something'
-    trade.update(limit_sell_order)
+    trade.update(limit_sell_order_usdt)
     assert trade.open_order_id is None
-    assert trade.close_rate == 0.00001173
-    assert trade.close_profit == 0.06201058
+    assert trade.close_rate == 2.20
+    assert trade.close_profit == round(0.0945137157107232, 8)
     assert trade.close_date is not None
     assert log_has_re(r"LIMIT_SELL has been fulfilled for Trade\(id=2, "
-                      r"pair=ETH/BTC, amount=90.99181073, open_rate=0.00001099, open_since=.*\).",
+                      r"pair=ETH/BTC, amount=30.00000000, "
+                      r"is_short=False, leverage=1.0, open_rate=2.00000000, open_since=.*\).",
+                      caplog)
+    caplog.clear()
+
+    trade = Trade(
+        id=226531,
+        pair='ETH/BTC',
+        stake_amount=60.0,
+        open_rate=2.0,
+        amount=30.0,
+        is_open=True,
+        open_date=arrow.utcnow().datetime,
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        is_short=True,
+        leverage=3.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
+    trade.update(limit_buy_order_usdt)
+    assert log_has_re(r"LIMIT_BUY has been fulfilled for Trade\(id=226531, "
+                      r"pair=ETH/BTC, amount=30.00000000, "
+                      r"is_short=True, leverage=3.0, open_rate=2.00000000, open_since=.*\).",
                       caplog)
```

Comment:
Would we want a corresponding test for sell below here (The unleveraged part tests both buy and sell updating ...)?